### PR TITLE
Add dark mode

### LINKED ISSUE #1146: Add a darkmode
Currently, https://plugins.jenkins.io/ has no dark mode at all. It would be pretty awesome, if we would have a darkmode switch on the page, allowing the user to toggle the theme or rely on the system theme and switch automatically.

### PR DESCRIPTION
**Pull request by zbynek**
Fixes #1146 

Before merging the header component should be updated.

### CODE REVIEW

File: plugins/plugin-site/gatsby-config.mjs [hunk 1]
```diff
@@ -28,6 +29,9 @@ Object.keys(process.env).forEach(key => {
     element[splits.slice(-1)[0]] = process.env[key];
 });
 
+const darkCodeTheme = fs.readFileSync('../../node_modules/github-syntax-dark/lib/github-dark.css', 'utf8');
```
Comment (halkeye) [inline plugins/plugin-site/gatsby-config.mjs]: can this be done with require.resolve? or import.meta.resolve? https://stackoverflow.com/questions/54977743/do-require-resolve-for-es-modules

seeing node_modules in paths always make me nervous, and .../.../node_modules doubly so.

File: plugins/plugin-site/src/styles/base.css [hunk 2]
```diff
@@ -992,3 +995,65 @@ a.card:hover {
...
+    --bs-highlight-color: #dee2e6;
+    --bs-highlight-bg: #664d03;
+    --bs-border-color: #495057;
+    --bs-border-color-translucent: rgba(255, 255, 255, 0.15);
+    --bs-form-valid-color: #75b798;
+    --bs-form-valid-border-color: #75b798;
+    --bs-form-invalid-color: #ea868f;
+    --bs-form-invalid-border-color: #ea868f;
+  }
+
```
Comment (halkeye) [inline plugins/plugin-site/src/styles/base.css]:
```suggestion
```

File: plugins/plugin-site/src/styles/base.css [hunk 3]
```diff
@@ -992,3 +995,65 @@ a.card:hover {
     max-width: 700px;
   }
 }
+@media (prefers-color-scheme: dark) {
```
Comment (halkeye) [inline plugins/plugin-site/src/styles/base.css:994]: how does this work for non dark? all the variables are already defined? or just empty?

Comment (zbynek) [reply plugins/plugin-site/src/styles/base.css:994]: All of them are defined, either in jenkins.io stylesheet or Bootstrap. The bootstrap values are taken from Bootstrap's dark mode (with link color modified).

Comment (zbynek) [reply plugins/plugin-site/src/styles/base.css:998]: All of them are defined, either in jenkins.io stylesheet or Bootstrap. The bootstrap values are taken from Bootstrap's dark mode (with link color modified).

**Review by timja — APPROVED** [review]
Looks good enough to me :) (tested in the deploy-preview), can be tweaked later.

Thanks for this!

**Comment by NotMyFault** [comment]
I'm late to the party but thanks for the PR!

I've tried it locally and found no issues so far. Looks really good

**Pull request merged**